src/fsck/fsck.c

```diff
@@ -172,10 +172,6 @@ static int process_progress(int fd) {
                 return log_debug_errno(errno, "Failed to use pipe: %m");
         }
 
-        console = fopen("/dev/console", "we");
-        if (!console)
-                return log_debug_errno(errno, "Failed to open /dev/console, can't print progress output: %m");
-
```

Comment:
If `console == NULL`, this function will crash below.

Reply:
I see now: this function should be called with console==NULL only when fd==-1, and in that case it return early. OK.